### ISSUE TITLE
No system Python installation found for Python 3.14

### ISSUE THREAD

**Issue by dcorto** [ansible/ansible-lint#4939]
##### Summary

No system Python installation found for Python 3.14

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)

Run cd $GITHUB_ACTION_PATH
  cd $GITHUB_ACTION_PATH
  uv pip install --system "ansible-lint[lock] @ git+https://github.com/ansible/ansible-lint@$GH_ACTION_REF"
  ansible-lint --version
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    ACTION_REF: main
    UV_PYTHON_INSTALL_DIR: /home/runner/work/_temp/uv-python-dir
    UV_PYTHON: 3.14
    UV_CACHE_DIR: /home/runner/work/_temp/setup-uv-cache
    GH_ACTION_REF: main
error: No system Python installation found for Python 3.14

```

##### STEPS TO REPRODUCE

```yml

# Workflow for lint code project
name: Lint

on:
  pull_request:

permissions:
  contents: read

jobs:
  ansible-lint:
    timeout-minutes: 5
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v6

      - name: Run ansible-lint
        uses: ansible/ansible-lint@main
        with:
          setup_python: "true"
          working_directory: "infrastructure/deploy"

```

##### Desired Behavior

Should work in Github actions

##### Actual Behavior

Do not works in ubuntu-latest

**Comment by konstruktoid** [comment]
Fails: https://github.com/konstruktoid/ansible-role-docker-rootless/actions/runs/21436985882/job/61730029339?pr=724 (commit https://github.com/ansible/ansible-lint/commit/64dfcc1dd96d681e3b0c684d5736bbb340e8b8d2) 
Succeeds: https://github.com/konstruktoid/ansible-role-hardening/actions/runs/21430772424/job/61709364029 (commit ff636f78696c40dd7037b591fbccb23abdcfc5ca)

**Comment by konstruktoid** [comment]
Workaround before fix:

```yaml
- name: Lint Ansible Playbook
  uses: ansible/ansible-lint@ff636f78696c40dd7037b591fbccb23abdcfc5ca
```

**Comment by dcorto** [comment]
thanks for the quick fix ❤️

**Comment by sdodsley** [comment]
@shatakshiiii @Jkhall81 I'm not sur ethis has fully fixed the issue.
```
Run ansible/ansible-lint@main
Run if [[ -n "" ]]; then
Run action_ref="${OVERRIDE_VER:-${OLD_REF:-${GITHUB_ACTION_REF_DISAMBIGUATED:-main}}}"
Using ansible-lint ref: main
Run reqs_file=$(git rev-parse --show-toplevel)/.git/ansible-lint-requirements.txt
--2026-01-28 15:12:13--  https://raw.githubusercontent.com/ansible/ansible-lint/main/.config/requirements-lock.txt
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.109.133, 185.199.110.133, 185.199.111.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.109.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1829 (1.8K) [text/plain]
Saving to: ‘/home/runner/work/FlashArray-Collection/FlashArray-Collection/.git/ansible-lint-requirements.txt’

     0K .                                                     100% 24.7M=0s

2026-01-28 15:12:13 (24.7 MB/s) - ‘/home/runner/work/FlashArray-Collection/FlashArray-Collection/.git/ansible-lint-requirements.txt’ saved [1829/1829]

Run uv tool install --python 3.14 "ansible-lint[lock] @ git+https://github.com/ansible/ansible-lint@$GH_ACTION_REF"
/home/runner/work/_temp/413b97ef-2a89-46cc-95d6-a6329ce67f83.sh: line 1: uv: command not found
Error: Process completed with exit code 127.
```
Don't we now nee `uv` as a pre-requisite?

**Comment by Jkhall81** [comment]
@ssbarnea I need some help figuring out this `uv: command not found Error`.  I tested this earlier locally with `act` and the action completed successfully.  But it seems to be failing on the GitHub runner.

**Comment by chriscarpenter12** [comment]
Similar issue for us. Seems like it was trying to use `uv` even though we have `setup_python: false`?

```
      - name: Run ansible-lint
        uses: ansible/ansible-lint@main
        with:
          args: ""
          setup_python: "false"
          requirements_file: requirements/galaxy.yml
```

```
Run reqs_file=$(git rev-parse --show-toplevel)/.git/ansible-lint-requirements.txt
--2026-01-28 19:44:21--  https://raw.githubusercontent.com/ansible/ansible-lint/main/.config/requirements-lock.txt
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.111.133, 185.199.108.133, 185.199.109.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.111.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1829 (1.8K) [text/plain]
Saving to: ‘/home/runner/work/openshift-ansible-deployment/openshift-ansible-deployment/.git/ansible-lint-requirements.txt’

     0K .                                                     100% 23.9M=0s

2026-01-28 19:44:21 (23.9 MB/s) - ‘/home/runner/work/openshift-ansible-deployment/openshift-ansible-deployment/.git/ansible-lint-requirements.txt’ saved [1829/1829]

Run uv tool install --python 3.14 "ansible-lint[lock] @ git+https://github.com/ansible/ansible-lint@$GH_ACTION_REF"
/home/runner/work/_temp/122215e5-c833-4b65-a886-284758807419.sh: line 1: uv: command not found
```

Reverted to the last tag and seems to be working

```
      - name: Run ansible-lint
        uses: ansible/ansible-lint@v26.1.1
        with:
          args: ""
          setup_python: "false"
          requirements_file: requirements/galaxy.yml
```

```
Run reqs_file=$(git rev-parse --show-toplevel)/.git/ansible-lint-requirements.txt
--2026-01-28 19:55:33--  https://raw.githubusercontent.com/ansible/ansible-lint/v26.1.1/.config/requirements-lock.txt
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.109.133, 185.199.110.133, 185.199.111.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.109.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1831 (1.8K) [text/plain]
Saving to: ‘/home/runner/work/openshift-ansible-deployment/openshift-ansible-deployment/.git/ansible-lint-requirements.txt’

     0K .                                                     100% 25.2M=0s

2026-01-28 19:55:33 (25.2 MB/s) - ‘/home/runner/work/openshift-ansible-deployment/openshift-ansible-deployment/.git/ansible-lint-requirements.txt’ saved [1831/1831]

Run cd $GITHUB_ACTION_PATH
Collecting ansible-lint@ git+https://github.com/ansible/ansible-lint@v26.1.1 (from ansible-lint[lock]@ git+https://github.com/ansible/ansible-lint@v26.1.1)
...
Run ansible-galaxy install -r requirements/galaxy.yml
Starting galaxy collection install process
Process install dependency map
Starting collection install process
...
Run exit_code=0

Passed: 0 failure(s), 0 warning(s) in 147 files processed of 250 encountered. Profile 'production' was required, and it passed.
```

**Comment by Jkhall81** [comment]
I was just looking at this..  So it looks like this is only happening if `setup_python` is set to false.  I guess we need to keep the pip install command in there, for the case where `setup_python` is false, so that this remains backwards compatible.  

Because right now, if `setup_python` is set to false, uv doesn't install, and then if there is no uv, we get this `uv: command not found` error.  My bad.  I will fix this.
